### ISSUE TITLE
Plugin errors for domain and datetime

### ISSUE THREAD

**Issue by pemontto** [fox-it/dissect.target#256]
I've installed dissect via pip a few different ways on different python version 3.9 through 3.11 and used the ghcr.io/fox-it/dissect:3.5 docker image. In all cases I get errors running target-info on both amd64 and aarch64.

Testing with some public E01 and vmdk files I'm getting plugin errors related to domain and datetime. Most other functions/commands seem to work however.

## [2020JimmyWilson.E01](https://drive.google.com/file/d/1Fd1pX1r4waRkD6Z2O8J5cRZyeSNU5-SY/view)
```bash
# target-shell 2020JimmyWilson.E01
2023-05-16T14:01:23.101142Z [warning  ] Failed to import <lazyattr dissect.target.loaders.asdf.AsdfLoader loaded=False> [dissect.target.loader]
2023-05-16T14:01:23.163181Z [warning  ] Failed to import <lazyattr dissect.target.volumes.bde.BitlockerVolumeSystem loaded=False> [dissect.target.volume]
2023-05-16T14:01:23.201831Z [warning  ] <Target 2020JimmyWilson.E01>: Can't identify filesystem: <Volume name='Microsoft reserved partition' size=33553920 fs=None> [dissect.target.target]
2023-05-16T14:01:23.202738Z [warning  ] Failed to import <lazyattr dissect.target.volumes.bde.BitlockerVolumeSystem loaded=False> [dissect.target.volume]
2023-05-16T14:01:23.492200Z [warning  ] <Target 2020JimmyWilson.E01>: Empty hive: sysvol/windows/system32/config/RegBack/SYSTEM [dissect.target.target]
IACIS-HDD-2014 /> domain
None
IACIS-HDD-2014 /> architecture
amd64-win64
IACIS-HDD-2014 /> hostname
IACIS-HDD-2014
IACIS-HDD-2014 /> os
windows
IACIS-HDD-2014 /> ips
['192.168.0.170']
IACIS-HDD-2014 /> install_date
2014-01-25 01:27:46+00:00
IACIS-HDD-2014 /> activity
2015-05-26 13:05:11.623703+00:00



# target-info 2020JimmyWilson.E01
2023-05-16T13:29:10.110702Z [warning  ] Failed to import <lazyattr dissect.target.loaders.asdf.AsdfLoader loaded=False> [dissect.target.loader]
2023-05-16T13:29:10.171415Z [warning  ] Failed to import <lazyattr dissect.target.volumes.bde.BitlockerVolumeSystem loaded=False> [dissect.target.volume]
2023-05-16T13:29:10.212305Z [warning  ] <Target 2020JimmyWilson.E01>: Can't identify filesystem: <Volume name='Microsoft reserved partition' size=33553920 fs=None> [dissect.target.target]
2023-05-16T13:29:10.213115Z [warning  ] Failed to import <lazyattr dissect.target.volumes.bde.BitlockerVolumeSystem loaded=False> [dissect.target.volume]
2023-05-16T13:29:10.508047Z [warning  ] <Target 2020JimmyWilson.E01>: Empty hive: sysvol/windows/system32/config/RegBack/SYSTEM [dissect.target.target]
<Target 2020JimmyWilson.E01>

Disks
- <EwfContainer size=891289600 vs=<DissectVolumeSystem serial=None>>

Volumes
- <Volume name='Microsoft reserved partition' size=33553920 fs=None>
- <Volume name='Basic data partition' size=855637504 fs=<NtfsFilesystem>>

Hostname      : IACIS-HDD-2014
Domain        : None
IPs           : 192.168.0.170
OS family     : windows (WindowsPlugin)
OS version    : Windows 7 Ultimate (NT 6.1) 7601 Service Pack 1
Architecture  : amd64-win64
Language(s)   :
2023-05-16T13:29:10.789079Z [error    ] <Target 2020JimmyWilson.E01>: Failed to load timezone information [dissect.target.target]
2023-05-16T13:29:10.789601Z [error    ] <Target 2020JimmyWilson.E01>: Exception in retrieving information for target: `<Target 2020JimmyWilson.E01>` [dissect.target.target]
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/dissect/target/target.py", line 587, in get_function
    self.add_plugin(plugin_cls)
  File "/usr/local/lib/python3.10/dist-packages/dissect/target/target.py", line 534, in add_plugin
    if p.check_compatible() is False:
  File "/usr/local/lib/python3.10/dist-packages/dissect/target/plugins/os/windows/datetime.py", line 208, in check_compatible
    raise UnsupportedPluginError("No time zone information")
dissect.target.exceptions.UnsupportedPluginError: No time zone information

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/dissect/target/tools/info.py", line 85, in main
    print_target_info(target)
  File "/usr/local/lib/python3.10/dist-packages/dissect/target/tools/info.py", line 131, in print_target_info
    print(f"Timezone      : {target.timezone}")
  File "/usr/local/lib/python3.10/dist-packages/dissect/target/target.py", line 642, in __getattr__
    result = func.__get__(p)
  File "/usr/local/lib/python3.10/dist-packages/dissect/target/helpers/cache.py", line 228, in cache_wrapper
    return cache.call(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/dissect/target/helpers/cache.py", line 104, in call
    func_cache[key] = self.func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/dissect/target/plugins/os/windows/locale.py", line 56, in timezone
    return normalize_timezone(self.target.datetime.tzinfo.name)
  File "/usr/local/lib/python3.10/dist-packages/dissect/target/target.py", line 637, in __getattr__
    p, func = self.get_function(attr)
  File "/usr/local/lib/python3.10/dist-packages/dissect/target/target.py", line 597, in get_function
    raise UnsupportedPluginError(
dissect.target.exceptions.UnsupportedPluginError: Unsupported function `datetime` for target with OS plugin <class 'dissect.target.plugins.os.windows._os.WindowsPlugin'>
```

## [ENISA-LOT3-Evidence.vmdk](https://www.enisa.europa.eu/ftp/ENISA-LOT3-Evidence.zip)
```bash
# target-info ENISA-LOT3-Evidence.vmdk
2023-05-16T13:28:52.464919Z [warning  ] Failed to import <lazyattr dissect.target.loaders.asdf.AsdfLoader loaded=False> [dissect.target.loader]
2023-05-16T13:28:52.535295Z [warning  ] Failed to import <lazyattr dissect.target.volumes.bde.BitlockerVolumeSystem loaded=False> [dissect.target.volume]
2023-05-16T13:28:52.814800Z [warning  ] <Target ENISA-LOT3-Evidence.vmdk>: Failed to find OS plugin, falling back to default [dissect.target.target]
<Target ENISA-LOT3-Evidence.vmdk>

Disks
- <VmdkContainer size=214748364800 vs=<DissectVolumeSystem serial=1068774149>>

Volumes
- <Volume name='part_00100000' size=214747316224 fs=<ExtFilesystem>>

Hostname      : None
2023-05-16T13:28:52.815804Z [error    ] <Target ENISA-LOT3-Evidence.vmdk>: Exception in retrieving information for target: `<Target ENISA-LOT3-Evidence.vmdk>` [dissect.target.target]
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/dissect/target/tools/info.py", line 85, in main
    print_target_info(target)
  File "/usr/local/lib/python3.10/dist-packages/dissect/target/tools/info.py", line 125, in print_target_info
    print(f"Domain        : {target.domain}")
  File "/usr/local/lib/python3.10/dist-packages/dissect/target/target.py", line 637, in __getattr__
    p, func = self.get_function(attr)
  File "/usr/local/lib/python3.10/dist-packages/dissect/target/target.py", line 605, in get_function
    raise PluginNotFoundError(f"Can't find plugin with function `{function}`")
dissect.target.exceptions.PluginNotFoundError: Can't find plugin with function `domain`
```

**Comment by pyrco** [comment]
I looked into the `TimeZoneKeyName` issue. It turns out the double null byte termination of utf-16 string typed registry values was not correctly handled. I made https://github.com/fox-it/dissect.regf/pull/19 to fix this.

As for the issue with the domain name, the plugins used to gather the info used in `target-info` are not available for each OS (only the functions in _os.py are guaranteed to be present), so I think it is a good idea to make target-info more robust against missing plugins.
